### DISCUSSION TITLE
Making minijinja-cli more useful

### DISCUSSION THREAD

**Discussion by mitsuhiko** [mitsuhiko/minijinja#620] [Ideas]
I find `minijinja-cli` as it stands a bit useless.  Yes, it can render some templates but you basically need to run some stuff in bash or else to populate the context.  I was thinking it might be interesting to add an option that provides some basic file system operations.

Potential options:

* `read_file`: reads a file from disk
* `read_dir`: enumerates files and folders in a folder
* `exec`: executes a program and return stdout/stderr/status

This obviously would require an option to enable.

Not sure if anyone has thoughts.

**Comment by marcodpt** [comment]
I'm suspicious of commenting.

but I consider it an excellent idea, and I would add http get requests to the list.

I'm betting on this type of approach to completely abandon backend languages ​​and javascript in the frontend.

minijinja + powerful filters and functions is the future!

**Comment by marcodpt** [comment]
@mitsuhiko 

Even in the current format I have made constant use of minijinja-cli, and I consider it far from useless.

My use is basically to synthesize data into a JSON file, and with this data generate SQL, cURL requests, and simple scripts.

You have a large SQL file for example that contains specific data mixed with the query logic, you separate the specific data into a JSON file and use minijinja-cli to generate the query.

Template engines have the wonderful ability to transform files from one format to another.

And minijinja is a brilliant implementation of that.
